packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -22,34 +23,62 @@ import {
   Unsubscribe,
 } from '@coveo/headless';
 import {InitializeEvent} from '../../utils/initialization-utils';
+import i18next, {i18n} from 'i18next';
+import Backend, {BackendOptions} from 'i18next-http-backend';
+import LanguageDetector, {
+  DetectorOptions,
+} from 'i18next-browser-languagedetector';
 
 @Component({
   tag: 'atomic-search-interface',
   shadow: true,
+  assetsDirs: ['assets'],
 })
 export class AtomicSearchInterface {
   @Element() host!: HTMLDivElement;
   @Prop() sample = false;
   @Prop({reflect: true}) pipeline = 'default';
   @Prop({reflect: true}) searchHub = 'default';
-  @Prop() logLevel?: LogLevel = 'info';
+  @Prop() logLevel?: LogLevel = 'silent';
+  @Prop() i18n: i18n = i18next.createInstance();
+  @Prop({mutable: true}) engine?: Engine;
   @State() error?: Error;
-  @State() engine?: Engine;
 
   private unsubscribe: Unsubscribe = () => {};
   private hangingComponentsInitialization: InitializeEvent[] = [];
   private initialized = false;
+  private afterInitializationCallbacks: (() => void)[] = [];
 
   componentWillLoad() {
     if (this.sample) {
       this.initialize(HeadlessEngine.getSampleConfiguration());
     }
+
+    this.i18n
+      .use(LanguageDetector)
```

Comment:
I don't think we should use the browser language to determine the localization of Atomic.

What's the desired outcome is for the search interface to have the same language as the rest of the website (which we cannot really know ahead of time).

For example, if my OS is french/Chrome is in french, but I go on help.salesforce.com where everything is in english, I don't expect to just see the search page in french.

We * could * try to detect some query parameter (like `?lang=fr` and stuff like this). 

However, for now, I would not try to do any magic, and instead rely on an option on the search interface component.

Reply:
It uses this order:
`order: ['querystring', 'cookie', 'localStorage', 'sessionStorage', 'navigator', 'htmlTag', 'path', 'subdomain']`
https://github.com/i18next/i18next-browser-languageDetector

Reply:
I have a function on the search interface already, but I like the idea of a watched "lang" prop better, will remove the LanguageDetector for now as it can be too much, but you gotta say it's a pretty good tool 😏